security/pkg/pki/ca/ca.go

```diff
@@ -275,15 +275,24 @@ type IstioCA struct {
 // NewIstioCA returns a new IstioCA instance.
 func NewIstioCA(opts *IstioCAOptions) (*IstioCA, error) {
 	ca := &IstioCA{
-		defaultCertTTL: opts.DefaultCertTTL,
-		maxCertTTL:     opts.MaxCertTTL,
-		keyCertBundle:  opts.KeyCertBundle,
-		livenessProbe:  probe.NewProbe(),
+		maxCertTTL:    opts.MaxCertTTL,
+		keyCertBundle: opts.KeyCertBundle,
+		livenessProbe: probe.NewProbe(),
 	}
 
 	if opts.CAType == selfSignedCA && opts.RotatorConfig.CheckInterval > time.Duration(0) {
 		ca.rootCertRotator = NewSelfSignedCARootCertRotator(opts.RotatorConfig, ca)
 	}
+
+	// if CA cert becomes invalid before workload cert it's going to cause workload cert to be invalid too,
+	// however citatel won't rotate if that happens, this function will prevent that using cert chain TTL as
+	// the workload TTL
+	defaultCertTTL, err := ca.getDefaultCertTTL(opts.DefaultCertTTL)
+	if err != nil {
+		return ca, fmt.Errorf("failed to get default cert TTL %s", err.Error())
```

Comment:
This is changing the original behavior. Before the change, the NewIstioCA() will always success and the defaultCertTTL will be the opts.DefaultCertTTL. After the change, it could fail if it failed to get the TTL from the cert.

Is this okay?

Reply:
I get your point, but I'm not really sure if it's ok to change this behavior (I think it is ok).